File: DayOffMini.Domain/DTOs/Reports/LeaveBalancesReportDto.cs
namespace DayOffMini.Domain.DTOs.Reports
{
    public class LeaveBalancesReportDto
    {
        public int EmployeeId { get; set; }
        public string EmployeeName { get; set; } = null!;

        public decimal TotalDaysOffRemaining { get; set; }
        public decimal TotalDaysOffBalance { get; set; }

        public ICollection<LeaveBalanceRowDto> LeaveBalances { get; set; } = new List<LeaveBalanceRowDto>();
    }
}

File: Frontend/src/app/reportCenter/models/LeaveBalancesReportDTO.ts
import { LeaveBalanceRowDTO } from './LeaveBalanceRowDTO';

export interface LeaveBalancesReportDTO {
  employeeId: number;
  employeeName: string;
  totalDaysOffRemaining: number;
  totalDaysOffBalance: number;
  leaveBalances: LeaveBalanceRowDTO[];
}
